samples/gen-js/index.js

```diff
@@ -0,0 +1,391 @@
+
+const discovery = require("@clever/discovery");
+const request = require("request");
+const url = require("url");
+const opentracing = require("opentracing");
+
+// go-swagger treats handles/expects arrays in the query string to be a string of comma joined values
+// so...do that thing. It's worth noting that this has lots of issues ("what if my values have commas in them?")
+// but that's an issue with go-swagger
+function serializeQueryString(data) {
+  if (Array.isArray(data)) {
+    return data.join(",");
+  }
+  return data;
+}
+
+module.exports = class SwaggerTest {
+
+  address;
+  timeout;
+
+  constructor(options) {
+    options = options || {};
+
+    if (options.discovery) {
+      this.address = discovery("swagger-test", "default").url();
+    } else if (options.address) {
+      this.address = options.address;
+    } else {
+      throw new Error("Cannot initialize swagger-test without discovery or address");
+    }
+    if (options.timeout) {
+      this.timeout = options.timeout
+    }
+  }
+
+  
+  getBooks = (params, options, cb) => {
+
+    if (!cb && typeof options === "function") {
+      cb = options;
+      options = undefined;
+    }
+
+    if (!options) {
+      options = {};
+    }
+
+    const timeout = options.timeout || this.timeout;
+    const span = options.span;
+
+    const headers = {
+
+    }
+
+    if (span) {
+      opentracing.inject(span, opentracing.FORMAT_TEXT_MAP, headers);
+      span.logEvent("GET /v1/books");
+    }
+
+    const requestOptions = {
+      method: "GET",
+      uri: this.address + "/v1/books",
+      json: true,
+      timeout,
+      headers,
+      body: {
+
+      },
```

Comment:
not sure how request handles this, but seems weird to define a body for a GET request

also feels bad in general to to define a request body for a method that doesn't take input from the request body